pootle/apps/pootle_store/updater.py

```diff
@@ -123,7 +123,8 @@ def should_create_suggestion(self):
     def should_update_index(self):
         return (
             self.update.change_indices
-            and self.uid in self.update.indices)
+            and self.uid in self.update.indices
+            and self.unit.index != self.update.get_index(self.uid))
```

Comment:
@phlax I'm recalling that we have allowed to have gaps in index ordering if they don't break a real sort order. Though I'm not sure how this change is related.